pkg/bundle/keychain.go

```diff
@@ -0,0 +1,77 @@
+package bundle
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sync"
+
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cli/cli/config/types"
+	"github.com/docker/docker/pkg/homedir"
+	"github.com/google/go-containerregistry/pkg/authn"
+)
+
+type podmanKeychain struct {
+	mu sync.Mutex
+}
+
+var PodmanKeyChain authn.Keychain = &podmanKeychain{}
+
+func (pk *podmanKeychain) Resolve(target authn.Resource) (authn.Authenticator, error) {
+	pk.mu.Lock()
+	defer pk.mu.Unlock()
+
+	authFile, err := os.Open(getPathToPodmanAuth())
+	// Return error only when the auth file is there but somehow unable to read.
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return authn.Anonymous, nil
+		}
+		return nil, err
+	}
+	defer authFile.Close()
+
+	cf, err := config.LoadFromReader(authFile)
+	if err != nil {
+		return nil, err
+	}
+
+	key := target.RegistryStr()
+	cfg, err := cf.GetAuthConfig(key)
+	if err != nil {
+		return nil, err
+	}
+
+	empty := types.AuthConfig{}
+	if cfg == empty {
```

Comment:
nit: you don't need to go by the empty variable i guess

Reply:
Indeed.  `if cfg == types.AuthConfig{}` is enough.  I used `empty` variable here more like for comment.